vendor/go.etcd.io/etcd/client/v3/kubernetes.go

```diff
@@ -0,0 +1,176 @@
+package clientv3
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+)
+
+func NewKubernetes(c *Client) Kubernetes {
+	return &kubernetes{kv: RetryKVClient(c), watcher: c}
+}
+
+type Kubernetes interface {
+	Get(ctx context.Context, key string, revision int64) (kv *mvccpb.KeyValue, rev int64, err error)
+	List(ctx context.Context, prefix string, revision, limit int64, fromKey string) (kvs []*mvccpb.KeyValue, count, rev int64, err error)
+	Count(ctx context.Context, prefix string) (int64, error)
+	OptimisticCreate(ctx context.Context, key string, value []byte) (success bool, rev int64, err error)
+	OptimisticUpdate(ctx context.Context, key string, value []byte, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+	OptimisticDelete(ctx context.Context, key string, expectedRevision int64) (kv *mvccpb.KeyValue, success bool, rev int64, err error)
+
+	Watch(ctx context.Context, key string, rev int64, prefix bool) WatchChan
+	RequestProgress(ctx context.Context) error
```

Comment:
Is this per watch progress or per etcd? If the latter, shouldn't it return something (like the current revision)? How does it work?

Reply:
> Is this per watch progress or per etcd? 

Yes, but it's per watch stream. Access to watch streams are not exposed to client, they are based on grpc metadata provided on the context. This usually means all watches use the same stream as long they don't override metadata for the grpc. If we are already changing etcd client we might consider exposing this. Added a `WatchID` response.

>  If the latter, shouldn't it return something (like the current revision)? 

Revision is returned to the watcher on the watch channel as it needs to be synchronized to other on watch events.